### PR TITLE
fix: deduplicate move operations

### PR DESCRIPTION
**Pull request by wraithgar**
Closes: https://github.com/npm/statusboard/issues/674

### CODE REVIEW

File: lib/content/write.js [hunk 1]
```diff
@@ -15,6 +15,9 @@ const fsm = require('fs-minipass')
 
 module.exports = write
 
+// Cache of move operations in process so we don't duplicate
+const moveOperations = {}
```
Comment (nlf) [inline lib/content/write.js:19]: my one and only piece of feedback for this is questioning if this is better as an object or a Map. it should get a lot of changes on the object, so i _think_ we would benefit from the faster key lookups and deletions in a Map

**Review by nlf — DISMISSED** [review]
marking as requested changes until we can hash out if a Map or object is better here

File: lib/content/write.js (hunk 1 above)
Comment (wraithgar) [reply lib/content/write.js]: Yep a map totally makes sense here. They're slightly slower to instantiate but we do that as a singleton so it's immeasurably small.  The repeated lookups will more than pay for themselves.

**Review by fritzy — CHANGES_REQUESTED** [review]

File: lib/content/write.js [hunk 2]
```diff
@@ -159,15 +162,23 @@ async function makeTmp (cache, opts) {
 async function moveToDestination (tmp, cache, sri, opts) {
   const destination = contentPath(cache, sri)
   const destDir = path.dirname(destination)
-
-  await fs.mkdir(destDir, { recursive: true })
+  if (moveOperations.has(destination)) {
+    return moveOperations.get(destination)
+  }
```
[the remark this answers is not in the document]
Comment (wraithgar) [reply lib/content/write.js:167]: No it wouldn't. We only return it if we didn't set the const ourselves.

[the remark this answers is not in the document]
Comment (wraithgar) [reply lib/content/write.js:167]: It's also a bit of defense in depth, I don't want to give us the opportunity to accidentally await it before we add it to the map.  Doing that would make this whole thing not work properly.

**Review by fritzy — APPROVED** [review]

**Pull request merged**